### DISCUSSION TITLE
USB connection from Santoker BT USB not working in 2.6

### DISCUSSION THREAD

**Discussion by generalstrikecoffee** [artisan-roaster-scope/artisan#862] [Q&A]
Hi all, please forgive my lack of technical knowledge in advance. Your patience is much appreciated :)

I'm roasting on a Santoker r500 which has a built in USB connection to connect to the computer. Artisan has this model of roaster in the "Machine" menu. When this machine is selected, the "Meter" in Device Assignment gets set to "Center 301", so I assume the Santoker has the same technology as a Center 301 built in, plus a USB output.  When I got this roaster (used) a few months ago I already had the drivers installed for a Center 304 that I was using with my old Huky roaster, so it worked right away and I've had no problems. Computer is a 2014 Macbook Air running OS Big Sur 11.6.5.

The other day I installed the newest version of Artisan and it stopped recognizing the USB signal from my roaster. Now when I select the machine, it only gives me the option for COM3 and not the Silicon Labs CP210x which I'm pretty sure was an option before (or it could have been another driver but I can't remember). 

SO, I panicked and wiped the whole computer, installed OS from scratch, installed the newest version of Artisan and the CP210x driver and nothing else, just to be sure it wasn't something else on the computer interfering. Still no luck getting Artisan to recognize the driver. Maybe I'm way off target here. I also ordered a Phidget + Vint hub to reroute the thermocouples if necessary (in case the USB interface on the roaster is the actual culprit coincidentally) but I thought I'd see if I'm missing something really obvious here. 

I also attempted to install the prior version of Artisan and the legacy CP210x driver but had the same problem. Any advice much appreciated! Thank you! <3 Kevin

**Comment by MAKOMO** [comment]
This is not an Artisan issue, but an issue with getting the correct driver installed (early CP210x drivers are not compatible with Big Sur, only the latest one) and recognising the hardware. Note that the port is only generated the moment the hardware is powered and connected.

**Reply by generalstrikecoffee** [reply]
Thank you! I have the latest CP210x driver installed, and the roaster connected and powered on, but no luck getting the driver to show up as an option.

**Reply by MAKOMO** [reply]
Then it might be a hardware or USB wire issue

**Reply by poundy** [reply]
I'm far from a Mac or Linux person... If I was troubleshooting this on Windows I'd use Device Manager to find what devices were present at the OS (taking away the application from the picture).  Can you look at /dev/tty* and see what are registered? 
ls -l /dev/tty*
That may get you a picture what can be seen by the OS... and I'd have the roaster disconnected and see the "before" view, then connect it and see if there's any change - if no different, as Marko says it points more to hardware/cables, but if a new port pops up then that somewhat eliminates that.
